util/client/ngmclient/client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package ngmclient
+
+import (
+	"context"
+	"fmt"
+	"net/http/httputil"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joomcode/errorx"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/fx"
+	"golang.org/x/sync/singleflight"
+
+	"github.com/pingcap/tidb-dashboard/pkg/utils/topology"
+	"github.com/pingcap/tidb-dashboard/util/topo"
+)
+
+var (
+	NgmErrNS                 = errorx.NewNamespace("error.api.ng_monitoring")
```

Comment:
Our new error ns in `util` is meant to be as short and expressive as possible:

```suggestion
	ErrNS                 = errorx.NewNamespace("ngm_client")
```